spec/models/book_spec.rb

```diff
@@ -99,6 +99,84 @@
+    it "Determines how many books were created per year" do
+      expect(Book.count_per_year).to eq({ })
+
+      book = Book.create!(name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1))
+
+      expect(Book.count_per_year).to eq({ 2000 => 1})
+
+      book2 = Book.create!(name: "Freedom", isbn: 1919191919, created_at: Date.new(2021, 1, 1))
+
+      expect(Book.count_per_year).to eq({ 2000 => 1, 2021 => 1})
+
+      # I can reasonably expect there to be either no books, a book, or multiple books.
+      # This test covers all of those options. Is this enough?
+    end
+  end
+
+  describe ".count_in_years" do
+    it "Determines how many books were created per year out of a list of provided years" do
+
+      # When we have no books, the value for every year is zero
+      expect(Book.count_in_years(2000, 2019, 2020)).to eq({ 2000 => 0, 2019 => 0, 2020 => 0,})
+
+      Book.create!(name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1))
+
+      # Adding one book updates the value appropriately
+      expect(Book.count_in_years(2000, 2019, 2020)).to eq({ 2000 => 1, 2019 => 0, 2020 => 0,})
+
+      Book.create!(name: "Freedom", isbn: 1919191919, created_at: Date.new(2021, 1, 1))
+
+      # The creation of a new book outside of the specified years does not change anything
+      expect(Book.count_in_years(2000, 2019, 2020)).to eq({ 2000 => 1, 2019 => 0, 2020 => 0,})
+    end
+  end
+
+  describe ".per_month" do
+    it "Determines how many books were created each month during a given year" do
+      # When we have no books, the value for every month is zero
+      expect(Book.per_month(2000)).to eq({ "January" => 0, "February" => 0, "March" => 0, "April" => 0, "May" => 0, "June" => 0, "July" => 0, "August" => 0, "September" => 0, "October" => 0, "November" => 0, "December" => 0,})
+
+      Book.create!(name: "Emma", isbn: 9292929292, created_at: Date.new(2000, 1, 1))
+
+      # Adding one book updates the value appropriately
+      expect(Book.per_month(2000)).to eq({ "January" => 1, "February" => 0, "March" => 0, "April" => 0, "May" => 0, "June" => 0, "July" => 0, "August" => 0, "September" => 0, "October" => 0, "November" => 0, "December" => 0,})
+
+      Book.create!(name: "Freedom", isbn: 1919191919, created_at: Date.new(2000, 2, 1))
+
+      # Adding a second book updates the value appropriately
+      expect(Book.per_month(2000)).to eq({ "January" => 1, "February" => 1, "March" => 0, "April" => 0, "May" => 0, "June" => 0, "July" => 0, "August" => 0, "September" => 0, "October" => 0, "November" => 0, "December" => 0,})
+
+      # Add a book with an incorrect year does not affect hash
+      Book.create!(name: "Freedom2", isbn: 1919191918, created_at: Date.new(2001, 2, 1))
+
+      expect(Book.per_month(2000)).to eq({ "January" => 1, "February" => 1, "March" => 0, "April" => 0, "May" => 0, "June" => 0, "July" => 0, "August" => 0, "September" => 0, "October" => 0, "November" => 0, "December" => 0,})
+    end
+  end
```

Comment:
There's a lot of repetition here with regards to defining the months.

It detracts from the reader's ability to differentiate between the test cases.

Really as a reader I want to understand the difference between each assertion as easily as possible.

Can you work out a way that we can reduce the duplication in the code without reducing the accuracy of the tests?